### PR TITLE
Add ability to define search synonyms

### PR DESCRIPTION
**Pull request by dvanscott**
I found myself wanting to provide some additional search terms to find one of my commands. My initial idea was to start putting those words in the description, but it ended up creating a really bad user experience with random words in the description that's displayed in the search results.

This PR adds the ability to add additional search terms (called synonyms here, but open to other ideas) to any command. This provides a simple way for developers to give their users additional terms to find commands.

For context, here's an example of how (and why) you might want to use synonyms for a command:

```php
class ViewBillingSettings extends SpotlightCommand
{
    protected string $name = 'View Billing Settings';

    protected string $description = 'Update your billing settings';

    protected array $synonyms = [
        'subscription',
        'credit card',
        'payment',
    ];

    public function execute(Spotlight $spotlight): void
    {
        $spotlight->redirect('/settings/billing');
    }
}
```

This would allow for a user to type `credit card` and be show the View Billing Settings command in the search results.

### CODE REVIEW

File: resources/js/spotlight.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ window.LivewireUISpotlight = (config) => {
         resolvedDependencies: {},
 
         init() {
-            this.commandSearch = new Fuse(this.commands, {threshold: 0.3, keys: ['name', 'description']});
-            this.dependencySearch = new Fuse([], {threshold: 0.3, keys: ['name', 'description']});
+            this.commandSearch = new Fuse(this.commands, {threshold: 0.3, keys: ['name', 'description', 'synonyms']});
+            this.dependencySearch = new Fuse([], {threshold: 0.3, keys: ['name', 'description', 'synonyms']});
```
Comment (PhiloNL) [inline resources/js/spotlight.js:21]: I don't think the synonyms apply for the dependencies or did you ran into issues if you did not apply the synonyms properties for both Fuse instances?

**Review by PhiloNL — COMMENTED** [review]
Thanks for the PR looks like a great addition! 🙌 
Just one question about one of the Fuse instances, other than that everything seems good to go. 😄

File: resources/js/spotlight.js (hunk 1 above)
Comment (agentphoenix) [reply resources/js/spotlight.js:21]: @PhiloNL I wasn't quite sure if they needed to be added there are not. Now that I think about it, you're probably right that they don't need to be there. Let me just run a couple of tests and if everything checks out, I'll push an update.

Comment (dvanscott) [reply resources/js/spotlight.js:21]: @PhiloNL I wasn't quite sure if they needed to be added there are not. Now that I think about it, you're probably right that they don't need to be there. Let me just run a couple of tests and if everything checks out, I'll push an update.

**Comment by dvanscott** [comment]
@PhiloNL Pulled those out of the dependency search and updated the command stub. Should be all set!

**Comment by PhiloNL** [comment]
Thanks @dvanscott 🙌

**Pull request merged**